presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/QueryRewriter.java

```diff
@@ -395,4 +430,271 @@ private static List<Property> applyPropertyOverride(List<Property> properties, L
                 .map(entry -> new Property(new Identifier(entry.getKey()), entry.getValue()))
                 .collect(toImmutableList());
     }
+
+    private static class NonDeterministicColumnRewriter
+            extends AstVisitor<Node, Void>
+    {
+        private static class FunctionCallSubstitute
+        {
+            private final QualifiedName name;
+            private final List<Integer> originalArgumentIndices;
+
+            private FunctionCallSubstitute(QualifiedName name, List<Integer> originalArgumentIndices)
+            {
+                this.name = requireNonNull(name, "name is null");
+                this.originalArgumentIndices = ImmutableList.copyOf(requireNonNull(originalArgumentIndices, "originalArgumentIndices is null"));
+            }
+        }
+
+        private static final Map<QualifiedName, FunctionCallSubstitute> FUNCTION_CALL_SUBSTITUTE_LOOKUP = ImmutableMap.<QualifiedName, FunctionCallSubstitute>builder()
+                .put(QualifiedName.of("approx_percentile"), new FunctionCallSubstitute(QualifiedName.of("avg"), ImmutableList.of(1)))
+                .put(QualifiedName.of("approx_distinct"), new FunctionCallSubstitute(QualifiedName.of("count"), ImmutableList.of(1)))
+                .build();
```

Comment:
Are these two the one functions we are going to handle for the rewrite? Also `avg` is non-deterministic.....

Reply:
`avg` implementation is non-deterministic!?!? Can you suggest a better substitute? Maybe `sum`?

On the other hand, I'm running a custom version w/ these two substitutes on prod traffic for a couple of days and it didn't report non-determinism.